### PR TITLE
[master] Fix constructor value sending and contracts creating contracts

### PR DESCRIPTION
**Pull request by valeryz**
## Description

This PR fixes multiple problems related to contracts creating other contracts.

## Backward Compatibility

- [x] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [ ] **ready for review**

### Integration Test (Core Team)

- [ ] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: src/libData/AccountData/AccountStoreSC.tpp [hunk 1]
```diff
@@ -442,45 +445,66 @@ bool AccountStoreSC<MAP>::UpdateAccounts(const uint64_t& blockNum,
...
-                                 << contractAddress.hex());
-        error_code = TxnStatus::FAIL_CONTRACT_ACCOUNT_CREATION;
-        return false;
-      }
-      Account* contractAccount = this->GetAccount(contractAddress);
-      if (contractAccount == nullptr) {
-        LOG_GENERAL(WARNING, "contractAccount is null ptr");
-        error_code = TxnStatus::FAIL_CONTRACT_ACCOUNT_CREATION;
-        return false;
+      Account* contractAccount;
```
Comment (CSideSteve) [inline src/libData/AccountData/AccountStoreSC.tpp:471]: This whole Atomics Area of the Accountstore is another obscure area of code that needs some clarity, should be addressed in a future refactoring.

File: src/libData/AccountData/AccountStoreSC.tpp [hunk 2]
```diff
@@ -277,8 +277,7 @@ uint64_t AccountStoreSC<MAP>::InvokeEvmInterpreter(
         try {
           if (it->hasCode() && it->Code().size() > 0) {
             targetAccount->SetImmutable(
-                DataConversion::StringToCharArray("EVM" + it->Code()),
-                contractAccount->GetInitData());
+                DataConversion::StringToCharArray("EVM" + it->Code()), {});
```
Comment (CSideSteve) [inline src/libData/AccountData/AccountStoreSC.tpp:280]: Is it ever possible that a smart contract would want to modify the contents of code retaining the previous value of data, for instance an account could have both scilla contracts and evm contracts and potentially an evm contract could change the contents of the scilla code field which is just a string and preserver whatever value of data was currently there.

**Review by CSideSteve — APPROVED** [review]
Apart from the questions I raised, LGTM

File: src/libData/AccountData/AccountStoreSC.tpp (hunk 2 above)
Comment (valeryz) [reply src/libData/AccountData/AccountStoreSC.tpp:280]: There's just no "data" in EVM accounts.
There's only balance, nonce, and storage (root).  If/when we start mixing EVM/Scilla in the same contract, we'll have to deal with this, but then we can't just all it SetImmutable anymore.

**Pull request merged**